### PR TITLE
CXX-862 Don't throw on invalid element access by index or key

### PR DESCRIPTION
**Pull request by acmorrow**

### CODE REVIEW

**Comment by acmorrow** [comment]
Please note that this is a subtle behavior change and we will need to relnote it.

File: src/bsoncxx/array/view.cpp [hunk 1]
```diff
@@ -175,9 +175,9 @@ view::iterator view::find(std::uint32_t i) const {
 
     if (bson_iter_init_find(&iter, &b, key.c_str())) {
         return iterator(element(iter.raw, iter.len, iter.off));
-    } else {
-        return end();
     }
+
+    return end();
```
Comment (Machyne) [inline src/bsoncxx/array/view.cpp]: Unrelated? Same in the document view iterator.

**Comment by Machyne** [comment]
Small nit, but LGTM!

File: src/bsoncxx/array/view.cpp (hunk 1 above)
Comment (acmorrow) [inline src/bsoncxx/array/view.cpp]: Just clearer than else { return ... };

Comment (Machyne) [inline src/bsoncxx/array/view.cpp]: Agreed. :+1:

**Pull request merged**